src/main/scala/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.scala

```diff
@@ -340,26 +376,48 @@ class CollectDuplexSeqMetrics
   }
 
-  /** Updates all the various metrics for the UMIs contained on the given records. */
-  private def updateUmiMetrics(ssGroups: Seq[Seq[SamRecord]]): Unit = {
+  /**
+    * Updates all the various metrics for the UMIs contained on the given records.
+    *
+    * @param ssGroups a Seq of length one or two containing the one or two single stranded tag families
+    *                 that comprise a single double-stranded tag family.  If there are two single-stranded
+    *                 tag families, there is no guarantee of their ordering within the Seq.
+    */
+  private[umi] def updateUmiMetrics(ssGroups: Seq[Seq[SamRecord]]): Unit = {
+    // ab and ba are just names here and _don't_ imply top/bottom strand, just different strands
     val (ab, ba) = ssGroups match {
       case Seq(a, b) => (a, b)
       case Seq(a)    => (a, Seq.empty)
       case _         => unreachable(s"Found ${ssGroups.size} single strand families in a double strand family!")
     }
 
-    val umi1s = new ArrayBuffer[String]
-    val umi2s = new ArrayBuffer[String]
+    val umi1s = new ArrayBuffer[String] // ab UMI 1 (and ba UMI 2) sequences
+    val umi2s = new ArrayBuffer[String] // ab UMI 2 (and ba UMI 1) sequences
 
     ab.iterator.map(r => r[String](RX).split('-')).foreach { case Array(u1, u2) => umi1s += u1; umi2s += u2 }
     ba.iterator.map(r => r[String](RX).split('-')).foreach { case Array(u1, u2) => umi1s += u2; umi2s += u1 }
 
-    Seq(umi1s, umi2s).foreach { umis =>
+    val Seq(abConsensusUmi, baConsensusUmi) = Seq(umi1s, umi2s).map{ umis =>
       val consensus = callConsensus(umis)
       val metric    = this.umiMetricsMap.getOrElseUpdate(consensus, UmiMetric(umi=consensus))
       metric.raw_observations    += umis.size
       metric.unique_observations += 1
       metric.raw_observations_with_errors += umis.filterNot(_ == consensus).size
+      consensus
+    }
+
+    if (this.duplexUmiCounts) {
+      // Make both possible consensus duplex UMIs.  We want to normalize to the pairing/orientation that we'd
+      // see on an F1R2 read-pair. We can get this either by direct observation in the `ab` set of reads,
+      // or by seeing an F2R1 in the `ba` set of reads.  This logic will pick more or less arbitrarily if the
+      // group of reads doesn't consist of the expected all F1R2s in one set and all F2R1s in the other set.
+      val duplexUmis = Seq(s"$abConsensusUmi-$baConsensusUmi", s"$baConsensusUmi-$abConsensusUmi")
+      val duplexUmi  = if (ab.exists(r => r.firstOfPair && r.positiveStrand) || ba.exists(r => r.secondOfPair && r.positiveStrand)) duplexUmis(0) else duplexUmis(1)
+
+      val metric = this.duplexUmiMetricsMap.getOrElseUpdate(duplexUmi, DuplexUmiMetric(umi=duplexUmi))
+      metric.raw_observations             += ab.size + ba.size
+      metric.unique_observations          += 1
+      metric.raw_observations_with_errors += (ab.iterator ++ ba.iterator).map(r => r[String](RX)).count(umi => !duplexUmis.contains(umi))
```

Comment:
Unfortunately no.  In this case I want the full duplex UMIs (e.g. `ACG-GGT`), which is what `(ab.iterator ++ ba.iterator).map(r => r[String](RX))` yields.  `(umi1s ++ umi2s)` would give a seq/iterator over the _individual_ UMIs (e.g. `ACG`, `GGT`).